script.js

```diff
@@ -148,7 +148,15 @@ const newYouTube = {
 
   registerListeners() {
     document.addEventListener('yt-visibility-refresh', newYouTube.inject); // Inject the button on info panel render.
-    document.addEventListener('yt-page-data-updated', newYouTube.injectCommentsCount); // Asynchronously get comments count when navigated to video page
+
+    // Asynchronously get comments count when navigated to video page.
+    document.addEventListener('yt-page-data-updated', function(e) {
+      // setTimeout mean: This event sometimes precedes rewriting of number of comment node. So wait few ms.
+      setTimeout(() => {
+        newYouTube.injectCommentsCount(e)
+      }, 500)
+    });
+
```

Comment:
Case of issue: heart-action typed video watched then next another video.
The next video's parse target comments node is rewritten late.